protocols/ledger/ledger.go

```diff
@@ -32,39 +32,44 @@ func (l *LedgerManager) HandleRequest(ledger *channel.TwoPartyLedger, request pr
 	if err != nil {
 		return protocols.SideEffects{}, fmt.Errorf("error finding a supported state: %w", err)
 	}
-
-	asset := types.Address{} // todo: loop over request.amount's assets
 	nextState := supported.Clone()
+	nextState.Outcome = outcome.Exit{}
+	// TODO: We should iterate over assets in RightAmount as well. For now we rely on the assumption that LeftAmount and RightAmount will have the same assets.
+	for asset := range request.LeftAmount {
+
+		currentLeftAmount := supported.Outcome.TotalAllocatedFor(request.Left)[asset]
+		currentRightAmount := supported.Outcome.TotalAllocatedFor(request.Right)[asset]
+		// Calculate the amounts
+		leftAmount := big.NewInt(0).Sub(currentLeftAmount, request.LeftAmount[asset])
+		rightAmount := big.NewInt(0).Sub(currentRightAmount, request.RightAmount[asset])
+		total := big.NewInt(0).Add(request.LeftAmount[asset], request.RightAmount[asset])
+		if leftAmount.Cmp(big.NewInt(0)) < 0 {
+			return protocols.SideEffects{}, fmt.Errorf("Allocation for %x cannot afford the amount %d", request.Left, request.LeftAmount[asset])
+		}
+		if rightAmount.Cmp(big.NewInt(0)) < 0 {
+			return protocols.SideEffects{}, fmt.Errorf("Allocation for %x cannot afford the amount %d", request.Right, request.RightAmount[asset])
+		}
 
-	// Calculate the amounts
-	amountPerParticipant := big.NewInt(0).Div(request.Amount[asset], big.NewInt(2))
-	leftAmount := big.NewInt(0).Sub(nextState.Outcome.TotalAllocatedFor(request.Left)[asset], amountPerParticipant)
-	rightAmount := big.NewInt(0).Sub(nextState.Outcome.TotalAllocatedFor(request.Right)[asset], amountPerParticipant)
-	if leftAmount.Cmp(big.NewInt(0)) < 0 {
-		return protocols.SideEffects{}, fmt.Errorf("Allocation for %x cannot afford the amount %d", request.Left, amountPerParticipant)
+		newOutcome := outcome.SingleAssetExit{
```

Comment:
Since `LedgerRequests` get handled outside of the `virtualfund` protocol I think it's possible that the ledger channel's outcome could change in the time between the request is generated and the request is fulfilled by the `LedgerManager`. I think that means we want to avoid the `LedgerRequest` containing a specific `SingleAssetExit` which could become outdate by the time we go to handle the request?

Reply:
Yeah you're right. I think this comment was caused by my same confusion as one of my other comments!